Description: minkipc: set idlc binary permission only on download
 Not all build enviornment has permissions to update the permissions
 of the installed/placed mink-idl-compiler "idlc" binary. One of them
 being Debian sbuild enviornment.

 Hence, set the permissions only when downloaded from the mink-idlc
 github archive.
Author: Abhinaba Rakshit <abhinaba.rakshit@oss.qualcomm.com>
Origin: upstream, https://github.com/qualcomm/minkipc/pull/59
Last-Update: 2026-04-30
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
Index: pkg-minkipc/CMakeLists.txt
===================================================================
--- pkg-minkipc.orig/CMakeLists.txt
+++ pkg-minkipc/CMakeLists.txt
@@ -33,10 +33,9 @@ if ("${MINKIDLC_BIN_DIR}" STREQUAL "${CM
 	https://github.com/quic/mink-idl-compiler/releases/download/v0.2.0/idlc
 	${MINKIDLC_BIN_DIR}/idlc
 	)
+	file(CHMOD ${MINKIDLC_BIN_DIR}/idlc PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ)
 endif()
 
-file(CHMOD ${MINKIDLC_BIN_DIR}/idlc PERMISSIONS OWNER_EXECUTE OWNER_WRITE OWNER_READ)
-
 add_compile_options(
 	-Wall -Wextra -Werror -Wshadow -Wcast-align
 	-Wmissing-declarations -Wformat-security -Wmissing-noreturn
